book/src/app.md

```diff
@@ -1,54 +1,33 @@
 # Application
 
-`App` define the type of note (e.g. XAN, ETH, BTC). Each application is identified by an address `appAddress` (the same way as user address identifies a user) and has its own VP `AppVP`.
+### Application logic
+Application VP is the defining structure of an application that contains the application logic. 
+Every time the application state is about to change, the application VP is called to validate the state transition.
+If the application VP doesn't consider a state transition valid, the state will not be changed.
 
-### Application VP
-Each application has its own [validity predicate](./validity-predicates.md) `AppVP` that defines the conditions on which the application can be sent/received/etc (e.g. whitelist VP that only allows using the application a specified set of users). As with other VPs, `AppVP` checks that input and output notes of the tx satisfy certain constraints.
-It is required that the `AppVP` of the applications involved in a tx evaluated to `true`.
+An application VP might require validity of other VPs, enforcing the VP hierarchy with the application VP being the main VP checked.
```

Comment:
It'd be worth mentioning at some point how the subVPs are checked. 

From previous discussions, this is done by checking the hash of the subVP inside the appVP and enforcing its validity separately.

Reply:
I'm not sure that this level of detalization aligns with the rest of the overview. I tried to remove the low-level details, and this, I believe, more of an implementation detail. But I agree that we need to mention it, but perhaps in the spec